builtInFunctions/setGuardian.go

```diff
@@ -0,0 +1,96 @@
+package builtInFunctions
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+const noOfArgsSetGuardian = 2
+const serviceUIDMaxLen = 32
+
+// SetGuardianArgs is a struct placeholder for all necessary args
+// to create a NewSetGuardianFunc
+type SetGuardianArgs struct {
+	BaseAccountGuarderArgs
+}
+
+type setGuardian struct {
+	baseActiveHandler
+	*baseAccountGuarder
+}
+
+// NewSetGuardianFunc will instantiate a new set guardian built-in function
+func NewSetGuardianFunc(args SetGuardianArgs) (*setGuardian, error) {
+	base, err := newBaseAccountGuarder(args.BaseAccountGuarderArgs)
+	if err != nil {
+		return nil, err
+	}
+	setGuardianFunc := &setGuardian{
+		baseAccountGuarder: base,
+	}
+	setGuardianFunc.activeHandler = args.EnableEpochsHandler.IsSetGuardianEnabled
+
+	return setGuardianFunc, nil
+}
+
+// ProcessBuiltinFunction will process the set guardian built-in function call
+func (sg *setGuardian) ProcessBuiltinFunction(
+	acntSnd, acntDst vmcommon.UserAccountHandler,
```

Comment:
in a set of built in funcitons, where we verifiy that sender == receiver we do not use the acntDst at all. Check ESDTNFTCreate for example

Reply:
refactored & simplified the code here https://github.com/multiversx/mx-chain-vm-common-go/pull/193